### DISCUSSION TITLE
Normal scripts "<script>" not working after page navigation

### DISCUSSION THREAD

**Discussion by emondhar** [vercel/next.js#32880] [Help] [answered]
After the page navigation my scripts is not loading anymore.

**Comment by jamesmosier** [comment, marked as answer]
Where and how are you loading the script?

**Reply by emondhar** [reply]
Hello James, 
I am using Script tag from 'next/script' on the index.js file      
```
<Script src="/js/f1.js" />
<Script src="/js/f2.js" />
```
Thank you for your time!

**Reply by emondhar** [reply]
I found this on the web : https://github.com/vercel/next.js/issues/4477.
But it didn't helped with my problem.

This one look promising : https://github.com/vercel/next.js/issues/4477#issuecomment-896885767, but I really want to use Next Link. 

**Is there a way to use external js file while browsing different pages?**

**Reply by jamesmosier** [reply]
I wasn't able to reproduce. If you have a full example you could share that'd be helpful.

**Reply by emondhar** [reply]
OK here a full exemple:
index.js
```
import Script from 'next/script'
import Header from '../components/Header'

export default function Home() {
  return (
    <div>

      <Header />

  
    </div>
  )

}
```
page2.js
```
import Script from 'next/script'
import Header from '../components/Header'

export default function Page2() {
  return (
    <div>

      <Header />

  
    </div>
  )

}
```

header.jsx
````
const Header = () => {
  return (
    <>
      <Script src="/js/imagesloaded.pkgd.min.js" />
      <Script src="/js/TweenMax.min.js" />
      <Script src="/js/ImagesPop.js" />

      <div className="frame">
        <h2 className="frame__pagetitle">
          Emon Dhar
        </h2>
      </div>
      <div className="content">
        <img className="content__img" src="img/1.jpg" alt="Some image" />
        <img className="content__img" src="img/2.jpg" alt="Some image" />
        <img className="content__img" src="img/3.jpg" alt="Some image" />

        <h3 className="content__title">EMON</h3>
      </div>
    </>
  );
};

export default Header;
````

**Reply by emondhar** [reply]
When I use nav link to route one page to another, my scripts are not reloading.

**Reply by jamesmosier** [reply]
It's because the script is already loaded, so the browser doesn't need to load it again. 

If you put a random query string on it you'll see it considered a different script url so it will load the script again (you can verify by checking the Network panel of your browser's devtools).

```
<Script
  src={`https://code.jquery.com/jquery-3.6.0.js?foo=${Math.round(
    Math.random() * 100
  )}`}
></Script>
```

**Reply by Kamud** [reply]
This worked @jamesmosier. It's a very clever workaround.

**Reply by pmarxbraun** [reply]
You should use [onReady](https://nextjs.org/docs/app/api-reference/components/script#onready) props of the Script component.
As the doc states

> Some third-party scripts require users to run JavaScript code after the script has finished loading and every time the component is mounted (after a route navigation for example).

**Comment by emondhar** [comment]
Thank you for your answer, it's helped me a lot!

**Comment by whitecodel** [comment]
I am also facing this issue but for the time being I just fixed this issue by importing all script code into useEffect here I am removing all events before my script code in useEffect

     useEffect(() => {
          $(document).ready(function () {
          var allElems = $('body').find('*');
          allElems.each(function () {
          var elm = this, //javascript DOM object
          $elm = $(this) //jquery object for element.
          $elm.off();
          // your script code here
       });
     });

**Comment by abdulsalamalmahdi** [comment]
I had the same problem in dev inv and prod inv . after implementing @jamesmosier or reloading scripts on route events, it worked in dev but when deploying to vercel it does not work

**Comment by ahmedyasser2222** [comment]
@jamesmosier  you can help me plz 
this scrept not work when make navigate 

> 'use client';
> import Script from 'next/script';
> import { useEffect } from 'react';
> 
> const MarketingBanner = () => {
>   useEffect(() => {
>     if (window.googletag && window.googletag.cmd) {
>       window.googletag.cmd.push(() => {
>         window.googletag.display('div-gpt-ad-1716715976462-0');
>       });
>     }
>   }, []);
> 
>   return (
>     <>
>       <Script src={url} />
>       <Script
>         id="google-ads-init"
>         dangerouslySetInnerHTML={{
>           __html: `
>               window.googletag = window.googletag || { cmd: [] };
>               googletag.cmd.push(function () {
>                 googletag.defineSlot('/34628225/new_search_2024_ad', [280, 345], 'div-gpt-ad-1716715976462-0').addService(googletag.pubads());
>                 googletag.pubads().enableSingleRequest();
>                 googletag.enableServices();
>               });
>             `,
>         }}
>       />
>     </>
>   );
> };
> 
> export default MarketingBanner;

**Reply by ahmedyasser2222** [reply]
@jamesmosier
